hibernate-core/src/main/java/org/hibernate/event/internal/DefaultRefreshEventListener.java

```diff
@@ -61,7 +63,23 @@ public void onRefresh(RefreshEvent event, RefreshContext refreshedAlready) {
 		final PersistenceContext persistenceContext = source.getPersistenceContextInternal();
 		final Object object = event.getObject();
 		if ( persistenceContext.reassociateIfUninitializedProxy( object ) ) {
-			if ( isTransient( event, source, object ) ) {
+			final boolean isTransient = isTransient( event, source, object );
+
+			if ( event.getLockOptions().getLockMode().greaterThan( LockMode.READ ) ) {
```

Comment:
IMO we should always load the entity graph with the refresh fetch profile, not just when a lock mode is specified. @sebersole can you please confirm that this if condition should be removed and loading should happen unconditionally?